### ISSUE TITLE
Publish functionality should have more verbose error messaging and have steps of validation changed

### ISSUE THREAD

**Issue by adgiles** [Esri/solutions-erg-widget#81]
Same issue as for GRG publishing: https://github.com/Esri/solutions-grg-widget/issues/175

When publishing if a user with access to an org, but not access to publish tries to create something, they get a generic message of "Create Service: Unable to create service". We should improve this message to tell them "Create Service: Unable to create service because user does not have permissions." Additionally we should change the workflow of validating the user's permissions. If a user with access to the org but not permissions to publish tries to create an item, using the name of an existing hosted feature service, it tells the user that they cannot publish because that name is not unique. This can be confusing for the user because it gives a false sense of what the problem is... true it cannot publish because the name is unique, but ultimately even if they then make the name unique, they still cannot publish because they do not have permissions, we should do the permissions check first, so they understand right away, that they cannot publish at all because they do not have permissions.

In short, the issue is: 
- [ ] improve message when user does not have permissions to "Create Service: Unable to create service because user does not have permissions."
- [ ] change order of validation to  
     1. Does user have access to org?
     2. Does user have permissions?
     3. Is name unique?

**Comment by adgiles** [comment]
@topowright 

Addressed in PR #82

Added a user permissions check before the layer name available check. If the user does not have publisher rights the following error message is displayed:

'Create Service: Unable to create service because user does not have permissions'

I think this use case is rarely going to happen as you must be a lv2 user within the portal to create a web application. Because of the nature of this widget is to create an authorised GRG layer for use across other applications, an application with this widget in it should not be shared to lv1 users.

So I think we have all bases covered.
